### PR TITLE
Repo reorg

### PR DESCRIPTION
**Pull request by grandizzy**
- contracts in root src dir
```
- src
	- commons
	- interfaces
		- pool
			- commons
			- erc20
			- erc721
		- position
		- rewards
	- libraries
		- external
		- helpers
		- internal
	Contracts
```
- updated analyze target and slither CI to analyze contracts from root src dir and external libraries from libraries/external

### CODE REVIEW

**Review by EdNoepel — COMMENTED** [review]
What about the Python stuff and `deploy.sh`?

**Comment by grandizzy** [comment]
> What about the Python stuff and `deploy.sh`?

Yikes, forgot about deploy, will fix

**Comment by grandizzy** [comment]
> What about the Python stuff and `deploy.sh`?

addressed in https://github.com/ajna-finance/contracts/pull/518/commits/645eed8aeba8cf10d29cbc6d71eb1dc8d01819e0, tested brownie test works without any other change

File: deploy.sh [hunk 1]
```diff
@@ -32,14 +32,13 @@ do
 done
 
 echo Deploying factories...
-factories=( 'erc20,ERC20PoolFactory' 'erc721,ERC721PoolFactory' )
+factories=( 'ERC20PoolFactory' 'ERC721PoolFactory' )
 for contract in "${factories[@]}"
 do
     IFS=',' read -r -a arr <<< ${contract}
```
Comment (EdNoepel) [inline deploy.sh]: We don't need the input field separator anymore if both factories are sitting in the same dir.

Comment (grandizzy) [reply deploy.sh]: pls check https://github.com/ajna-finance/contracts/pull/518/commits/987805d7a58c014985367cf1a3380951744abc1d

**Review by EdNoepel — APPROVED** [review]
👌

**Review by MikeHathaway — APPROVED** [review]
LGTM! Only suggestions would be to maybe rename `commons` to `base` signifying they're all abstract contracts. Commons also doesn't make sense for things like `PermitERC721` which is only used by `PositionManager`. Also having nested base/interfaces seems a little weird. Could those just be moved to the interfaces folder?

**Comment by grandizzy** [comment]
> LGTM! Only suggestions would be to maybe rename `commons` to `base` signifying they're all abstract contracts. Commons also doesn't make sense for things like `PermitERC721` which is only used by `PositionManager`. Also having nested base/interfaces seems a little weird. Could those just be moved to the interfaces folder?

renamed commons to base with https://github.com/ajna-finance/contracts/commit/5735ed33253c21e62f5c9a89ab9b9928fef59010

**Pull request merged**